component/core/src/main/java/org/exoplatform/social/core/metadata/thumbnail/ImageThumbnailServiceImpl.java

```diff
@@ -0,0 +1,108 @@
+
+  private final MetadataService     metadataService;
+
+  private final FileService         fileService;
+
+  private final ImageResizeService  imageResizeService;
+
+  public ImageThumbnailServiceImpl(MetadataService metadataService,
+                                   FileService fileService,
+                                   ImageResizeService imageResizeService) {
+    this.metadataService = metadataService;
+    this.fileService = fileService;
+    this.imageResizeService = imageResizeService;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public FileItem getOrCreateThumbnail(FileItem file, Identity identity, int width, int height) throws Exception {
+    if (file == null) {
+      throw new IllegalArgumentException("file argument is mandatory");
+    }
+    if (identity == null) {
+      throw new IllegalArgumentException("User identity argument is mandatory");
+    }
+    if (width == 0 && height == 0) {
+      return file;
+    }
+    FileInfo fileInfo = file.getFileInfo();
+    ThumbnailObject thumbnailObject = new ThumbnailObject(THUMBNAIL_OBJECT_TYPE, Long.toString(fileInfo.getId()));
+    List<MetadataItem> metadataItemList = metadataService.getMetadataItemsByMetadataTypeAndObject(THUMBNAIL_METADATA_TYPE.getName(),
+                                                                                                thumbnailObject);
+    List<MetadataItem> items = metadataItemList.stream()
+                                               .filter(metadataItem -> metadataItem.getProperties() != null
+                                                   && metadataItem.getProperties()
+                                                                  .get(THUMBNAIL_WIDTH_PROPERTY)
+                                                                  .equals(String.valueOf(width))
+                                                   && metadataItem.getProperties()
+                                                                  .get(THUMBNAIL_HEIGHT_PROPERTY)
+                                                                  .equals(String.valueOf(height)))
+                                               .collect(Collectors.toList());
+    if (!items.isEmpty()) {
+      long fileId = Long.parseLong(items.get(0).getParentObjectId());
+      try {
+        return fileService.getFile(fileId);
+      } catch (FileStorageException e) {
+        return null;
+      }
+    } else {
+      byte[] imageContent = imageResizeService.scaleImage(IOUtils.toByteArray(file.getAsStream()), width, height, false, false);
+      FileItem thumbnail = new FileItem(null,
+                                        fileInfo.getName(),
+                                        fileInfo.getMimetype(),
+                                        SOCIAL_NAME_SPACE,
+                                        imageContent.length,
+                                        new Date(),
+                                        identity.getRemoteId(),
```

Comment:
Can we say that the updater for the thumbnail is the same as the updater of the original file ?